Automattic/wp-calypso
# Add agnostic email-accounts data layer

#### Changes proposed in this Pull Request

*  A feature flag `email-accounts/enabled`
*  A new endpoint slug `email-accounts` to represent all email accounts per site regardless of the context of a provider
*  A new HTTP path  i.e. `/sites/${ siteId }/email-accounts`. This is not final and mostly a placeholder.
*  A new data component `QueryEmailAccounts` to manage fetching email accounts data. 

NOTE: This PR essentially gets the Redux boilerplate out of the way. The data structure is a work in progress and will most likely see further revisions in subsequent PRs.

#### Testing instructions



* Start your Calypso dev server and navigate to : http://calypso.localhost:3000/email
* Use the Redux profiler to ensure that a new key `emailAccounts` exists in the tree.
